src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -727,7 +727,7 @@ static CHIP_ERROR EncodeSigma1(MutableByteSpan & buf)
                                         ByteSpan(initiatorResumeMIC, Params::initiatorResumeMICLen)));
     }
 
-    if (Params::includeStructEnd)
+    if constexpr (Params::includeStructEnd)
```

Comment:
Why constexpr in all these ifs? It's test code. Not necessarily a bad thing, but currious why.

Reply:
I was just making someone's wish true (just above)
> // I wish we had "if constexpr" support here, so the compiler would know
    // resumptionIdLen is nonzero inside the block....